pkg/controllers/provisioning/provisioner.go

```diff
@@ -416,11 +416,144 @@ func (p *Provisioner) getDaemonSetPods(ctx context.Context) ([]*v1.Pod, error) {
 		pod := p.cluster.GetDaemonSetPod(&d)
 		if pod == nil {
 			pod = &v1.Pod{Spec: d.Spec.Template.Spec}
+		} else {
+			mergedAffinity := p.MergeAffinities(d.Spec.Template.Spec.Affinity, pod.Spec.Affinity)
```

Comment:
Does it make sense to merge the affinities or just use the DaemonSet node affinity instead of the node affinity that comes from the pod, since that one will always get overridden by the DaemonSet controller as shown here: https://github.com/kubernetes/kubernetes/blob/c5cf0ac1889f55ab51749798bec684aed876709d/pkg/controller/daemon/util/daemonset_util.go#L176.

I think that could help simplify some things. You basically take every detail of the pod, except for the Required Node Affinities which you expect to be overridden by the DaemonSet controller.

Reply:
I was attempting to do a thorough merge in the event that kubernetes behaviour ever changes again in the future, but i do agree that it could simplify some stuff too, let me make the change

Reply:
It seems unlikely to me that they will make a change there and (even if they did) we would just update our logic then when we see the logic changes in the new K8s version changelog